src/com/seafile/seadroid2/ui/activity/ShibbolethAuthorizeActivity.java

```diff
@@ -60,6 +73,34 @@ private void openAuthorizePage(String url) {
             return;
         }
 
+        String deviceId = Settings.Secure.getString(this.getContentResolver(),
+                Settings.Secure.ANDROID_ID);
+
+        String appVersion = "";
+        Context context = SeadroidApplication.getAppContext();
+        try {
+            PackageInfo pInfo = context.getPackageManager().
+                    getPackageInfo(context.getPackageName(), 0);
+            appVersion = pInfo.versionName;
+        } catch (PackageManager.NameNotFoundException e) {
+            // ignore
+        }
+
+        if (!url.endsWith("/")) {
+            url += "/shib-login";
+        } else
+            url += "shib-login";
+
+        url += String.format("?shib_platform_version=%s&shib_device_name=%s&shib_platform=%s&shib_device_id=%s&shib_client_version=%s",
+                URLEncoder.encode(Build.VERSION.RELEASE, "UTF-8"),
+                URLEncoder.encode(Build.MODEL, "UTF-8"),
+                "android",
+                URLEncoder.encode(deviceId, "UTF-8"),
+                appVersion);
```

Comment:
We can suppress the `UnsupportedEncodingException` here, since this exception is impossible here because all the params wold be plain ascii characters.